Library/Homebrew/build.rb

```diff
@@ -172,8 +172,7 @@ def fixopt f
 end
 
 begin
-  error_pipe = IO.new(ENV["HOMEBREW_ERROR_PIPE"].to_i, "w")
-  error_pipe.fcntl(Fcntl::F_SETFD, Fcntl::FD_CLOEXEC)
+  error_pipe = UNIXSocket.open(ENV["HOMEBREW_ERROR_PIPE"]).recv_io
```

Comment:
This patch introduces a file descriptor leak on ruby 1.8, where close-on-exec is not the default.

Add the following lines here:

```rb
system "ls", "/dev/fd"
exit
```

On the prior implementation:

```
$ brew install tree -s
0 1 2 3 4
```

(this is the expected set of FDs for `ls` to have open.)


With this patch:

```
 $ brew install tree -s
0 1 2 3 4 5 6 7 8
```